✨ Implement contracts of `business-on-behalf-product-list-connector`-bundle

daniel-rose opened the pull request:
- validate cart items by product list of default company user
- remove cart items by product list of default company user
- set default company user before validating or removing cart items
- ...

sourcecube commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -11,6 +11,7 @@
   "require": {
     "php": ">=7.3",
     "fond-of-oryx/one-time-password": "^1.0.0",
+    "spryker/customer": "^1.0.0 || ^2.0.0 || ^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0",
```
I might be unnecessary to support 7 major versions, 2 versions should be sufficient.

daniel-rose committed:
✨ Implement contracts of `business-on-behalf-product-list-connector`-bundle
- validate cart items by product list of default company user
- remove cart items by product list of default company user
- set default company user before validating or removing cart items
- ...

daniel-rose committed:
Add phpcs fixes
